accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,282 @@
+In this case, this tool is mean not only to provide a picture of the current health status of the Uyuni server, but also to help engineers to debug disconnected setups via "supportconfig". In this regard, the tool cannot always rely on existing Prometheus and Grafana instances, but it should be able to deploy its own instances to visualize the data on disconnected environments.
+
+- The Health Check Tool is not installed or running by default in a Uyuni server.
+- As multiple components (containers) are deployed, resources on a live Uyuni server might be affected.
+- It should be able to reuse existing Prometheus and Grafana instances, allowing integration with existing Monitoring stack.
+
+
+# TBD: Tentative plan & schedule
+
+### Phase 0 (before 5.0 GA):
+
+#### Uyuni Health Check
+- Create RPM packages in OBS/IBS.
+- Use container images building on OBS/IBS for the different components.
+- Provide an "standalone" version only (no integration with Monitoring stack). Via Uyuni server channel (or others so it is available for engineering or supporters).
+- Flag this as a Tech Preview.
+
+#### Saline
+- Research about using TCP instead of IPC sockets for Salt Master internal sockets.
+- Build Saline in OBS/IBS and provide it via RPM package for the Uyuni Server channel.
+- Saline runs inside "uyuni-server" container.
+
+### Phase 1 (after 5.0 GA):
+
+#### Uyuni Health Check
+- Enhance metrics, dashboards and alerts.
+- Allow user to enhance Loki pipeline to add filters and/or replace patterns to replace sensitive information.
+- Integration with Monitoring Stack. Enhance monitoring formula to enable Health Check monitoring.
+- Reuse configuration from "standalone" version.
+
+#### Saline
+- Provide "Saline" inside a separated container that attach to Salt Master TCP sockets from "uyuni-server" container.
+- Alternatively, explore sharing Salt Master IPC sockets between containers.
+
+
+# How is the exporter going to access to uyuni-server container? For example, to query the database.
+As mentioned, the exporter needs to have access to Uyuni server container. It does need to query the DB and also execute some Salt runners jobs to gather metrics. If the "exporter" runs in a separated container than the Salt Master, then we can make the "exporter" container to run in the same network than the "uyuni-server", then TCP sockets are available to use.
+
+There are some Salt runners jobs that requires access to Salt Master Event publisher, currently exposed via IPC socket. This makes it tricky access these sockets from a different container. Some metrics might not be available until we move to "TCP" sockets for the Salt Master Event publisher in this case.
+
+Alternatively, we could make the "exporter" as part of the server-image, even if not running by default. Then it is up to the Healh Check Tool or Monitoring formula to start it. This would solve the current issue of not having access to the IPC sockets.
```

Comment:
We have two parts. Monitoring formula takes care of exporters on the minions and does not run on the server.
We also have monitoring state on the server which is triggered/enabled via the API. It will need to be reworked in the containerized server anyway.